### PR TITLE
Posts: display no-ads notice only if there are more than three posts.

### PR DESCRIPTION
**Pull request by mtias**

### CODE REVIEW

**Comment by artpi** [comment]
Testing

**Comment by artpi** [comment]
🐑 🚢

**Comment by artpi** [comment]
I am a bit concerned here that we count impression of a nudge that is buried deep down at the bottom which can get us bad results

File: client/my-sites/posts/post-list.jsx [hunk 1]
```diff
@@ -224,19 +224,29 @@ var Posts = React.createClass( {
 		);
 	},
 
-	renderPost: function( post ) {
+	renderPost: function( post, index ) {
```
Comment (artpi) [inline client/my-sites/posts/post-list.jsx]: Infinitelist passes index arg:
https://github.com/Automattic/wp-calypso/blob/master/client/components/infinite-list/index.jsx#L356

**Comment by mcsf** [comment]
Drive-by thought: what happens when Headstart is enabled? I assume its _lorem ipsum_ posts count?

**Comment by mtias** [comment]
@mcsf they would, yes.

**Comment by mcsf** [comment]
OK, so that means this change is generally effect-less, NUX-wise, for Headstarted sites. :)

We'll need to think of something else for those scenarios. /cc @meremagee @shaunandrews

**Comment by mtias** [comment]
Does it matter much at this moment? The new positioning won't be the first thing you see anymore.

**Comment by mcsf** [comment]
> The new positioning won't be the first thing you see anymore.

A-ha, I glossed over that. That's an improvement, yeah.

**Comment by michaeldcain** [comment]
FWIW, there's a site option that gives the count of edited/unedited Headstart posts/pages, but I don't think it's available through an endpoint, so would involve a good bit of work. Given the new positioning of the nudge, it's probably not a big deal. Also, really only in themes with a grided post layout does Headstart add more than one default blog post, so it probably won't have much of an effect at this point.

File: client/my-sites/posts/post-list.jsx [hunk 2]
```diff
@@ -238,6 +237,22 @@ var Posts = React.createClass( {
 				path={ route.sectionify( this.props.context.pathname ) }
 			/>
 		);
+
+		if ( index === 2 && this.props.sites.getSelectedSite() && ! this.props.statusSlug ) {
+			return (
+				<div>
```
Comment (aduth) [inline client/my-sites/posts/post-list.jsx]: >warning.js:45 Warning: Each child in an array or iterator should have a unique "key" prop. Check the render method of `InfiniteList`. See https://fb.me/react-warning-keys for more information.

Comment (artpi) [inline client/my-sites/posts/post-list.jsx]: @aduth:
Thakns, investigating !

Comment (artpi) [inline client/my-sites/posts/post-list.jsx]: https://github.com/Automattic/wp-calypso/pull/5265

**Pull request merged**